### PR TITLE
Enable pass attributes to script and link

### PR DESCRIPTION
**Pull request by wszgxa**
## Motivation
Github issue: #848 

Enable pass attributes to script tag and link tag.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

I have tested this locally, it will apply the attributes to script and link.

## Related PRs

nope

### CODE REVIEW

**Comment by yangshun** [comment]
Retried Netlify deploy!

File: lib/core/Head.js [hunk 1]
```diff
@@ -134,13 +134,23 @@ class Head extends React.Component {
...
+            }
+            return <link rel="stylesheet" key={source} href={source} />;
+          })}
         {this.props.config.scripts &&
-          this.props.config.scripts.map(source => (
-            <script type="text/javascript" key={source} src={source} />
-          ))}
+          this.props.config.scripts.map(source => {
+            if (source instanceof Object) {
+              return source.src ? (
```
Comment (endiliey) [inline lib/core/Head.js]: Use [Inline If with Logical && Operator](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator)

e.g:
```jsx
return source.src && <script type="text/javascript" key={source.src} {...source} />
```

File: docs/api-site-config.md [hunk 2]
```diff
@@ -155,15 +155,15 @@ h1 {
 
 * `separate` - The secondary navigation is a separate pane defaulting on the right side of a document. See http://docusaurus.io/docs/en/translation.html for an example.
 
-`scripts` - Array of JavaScript sources to load. The script tag will be inserted in the HTML head.
+`scripts` - Array of JavaScript sources to load. You can pass only src link as srting, or you can pass attributes as object. The script tag will be inserted in the HTML head.
```
Comment (yangshun) [inline docs/api-site-config.md]: Typo in `srting`. Use this instead:

```
Array of JavaScript sources to load. The values can be either strings or plain objects of attribute-value maps. Refer to the example below. The script tag will be inserted in the HTML head.
```

File: docs/api-site-config.md [hunk 3]
```diff
@@ -155,15 +155,15 @@ h1 {
...
 
 `separateCss` - Directories inside which any `css` files will not be processed and concatenated to Docusaurus' styles. This is to support static `html` pages that may be separate from Docusaurus with completely separate styles.
 
 `scrollToTop` - Set this to `true` if you want to enable the scroll to top button at the bottom of your site.
 
 `scrollToTopOptions` - Optional options configuration for the scroll to top button. You do not need to use this, even if you set `scrollToTop` to `true`; it just provides you more configuration control of the button. You can find more options [here](https://github.com/vfeskov/vanilla-back-to-top/blob/v7.1.14/OPTIONS.md). By default, we set the zIndex option to 100.
 
-`stylesheets` - Array of CSS sources to load. The link tag will be inserted in the HTML head.
+`stylesheets` - Array of CSS sources to load. You can pass only href link as srting, or you can pass attributes as object. The link tag will be inserted in the HTML head.
```
Comment (yangshun) [inline docs/api-site-config.md]: Similar changes.

File: lib/core/Head.js [hunk 4]
```diff
@@ -134,13 +134,23 @@ class Head extends React.Component {
...
+                <link rel="stylesheet" key={source.href} {...source} />
+              ) : null;
+            }
+            return <link rel="stylesheet" key={source} href={source} />;
+          })}
         {this.props.config.scripts &&
-          this.props.config.scripts.map(source => (
-            <script type="text/javascript" key={source} src={source} />
-          ))}
+          this.props.config.scripts.map(source => {
```
Comment (yangshun) [inline lib/core/Head.js]: Better written as:

```jsx
this.props.config.scripts.map(script => {
  const scriptProps = Object.assign({}, 
    { type: 'text/javasript' }, 
    script instanceOf 'Object' ? script : { src: source },
  );
  return <script {...source} />;
});
```

File: lib/core/Head.js [hunk 5]
```diff
@@ -134,13 +134,23 @@ class Head extends React.Component {
 
         {/* External resources */}
         {this.props.config.stylesheets &&
-          this.props.config.stylesheets.map(source => (
-            <link rel="stylesheet" key={source} href={source} />
-          ))}
+          this.props.config.stylesheets.map(source => {
```
Comment (yangshun) [inline lib/core/Head.js]: Rewrote here in a similar fashion as well.

**Comment by wszgxa** [comment]
Will address comments at tonight.

File: lib/core/Head.js (hunk 4 above)
Comment (wszgxa) [reply lib/core/Head.js]: We need to include the key, 
``` js
this.props.config.scripts && 
  this.props.config.scripts.map(source => {
    const scriptProps = Object.assign(
        {type: 'text/javasript'},
        source.src ? source : {src: source} // include key in this line is more complexity than use ?..: ...
    );
    return <script {...scriptProps} />;
})
```

I guess if we don't need key, we can do as your example.

**Comment by wszgxa** [comment]
Hi, guys

I have addressed the comments. Because we need to pass down a unique key. 

I guess use ternary operator is more readable than use Object.assign.

**Pull request merged**